iModelCore/iModelPlatform/DgnCore/TxnManager.cpp

```diff
@@ -2004,6 +2018,7 @@ void TxnManager::DeleteAllTxns() {
     }
 
     m_dgndb.SaveChanges(); // in case there are outstanding changes that will create a new Txn
+    ReverseAll();

```

Comment:
This only reverse txn in current session. Again discardChanges() does full reverse down to first session.